src/docfx/lib/JsonUtility.cs

```diff
@@ -482,7 +482,32 @@ private static void SetFieldWritable(MemberInfo member, JsonProperty prop)
             }
         }
 
-        private sealed class JsonContractResolver : DefaultContractResolver
+        private class JsonContractResolver : CamelCasePropertyNamesContractResolver
+        {
+            protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+            {
+                var prop = base.CreateProperty(member, memberSerialization);
+                var contract = DefaultSerializer.ContractResolver.ResolveContract(prop.PropertyType);
```

Comment:
only do this if `ShouldSerialize` is null

Reply:
https://github.com/JamesNK/Newtonsoft.Json/blob/509643a8952ce731e0207710c429ad6e67dc43db/Src/Newtonsoft.Json/Serialization/CamelCasePropertyNamesContractResolver.cs#L69
There is a cache from type to JsonContract, the effort is minor